plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/parser/SQLParserContext.java

```diff
@@ -74,7 +94,14 @@ public IDocument getDocument() {
     }
 
     public SQLDialect getDialect() {
-        return SQLUtils.getDialectFromDataSource(getDataSource());
+        if (dataSourceContainer != null) {
+            try {
+                return dataSourceContainer.getScriptDialect().createInstance();
```

Comment:
I believe we shouldn't create dialect instance this way. Described in another comment to a similar place.

Reply:
At least we can cache it.